lib/evmone/baseline.cpp

```diff
@@ -106,76 +106,94 @@ inline evmc_status_code check_requirements(
 
 
 /// Implementation of a generic instruction "case".
-#define DISPATCH_CASE(OPCODE)                                               \
-    case OPCODE:                                                            \
-        ASM_COMMENT(OPCODE);                                                \
-        if (code_it = invoke<OPCODE>(cost_table, state, code_it); !code_it) \
-            goto exit;                                                      \
+#define DISPATCH_CASE(OPCODE)                                                            \
+    case OPCODE:                                                                         \
+        ASM_COMMENT(OPCODE);                                                             \
+                                                                                         \
+        if (const auto next = invoke<OPCODE>(cost_table, stack_bottom, position, state); \
+            next.code_it == nullptr)                                                     \
+        {                                                                                \
+            goto exit;                                                                   \
+        }                                                                                \
+        else                                                                             \
+        {                                                                                \
+            /* Update current position only when no error,                               \
+               this improves compiler optimization. */                                   \
+            position = next;                                                             \
+        }                                                                                \
         break
 
+
+/// The execution position.
+struct Position
```

Comment:
Nit: maybe move it above the macro, because `position` is used in it.

Reply:
Moved the macro down.